test/extended/clusterup.sh

```diff
@@ -0,0 +1,194 @@
+}
+
+function os::test::extended::clusterup::verify_router_and_registry () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get svc -n default" "docker-registry"
+    os::cmd::expect_success_and_text "oc get svc -n default" "router"
+    os::cmd::try_until_success "oc get endpoints docker-registry -n default | grep ':5000'" $(( 10*minute )) 1
+    os::cmd::try_until_success "oc get endpoints router -n default | grep ':80'" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_image_streams () {
+    os::cmd::try_until_success "oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }' | grep latest" $(( 10*minute )) 1
+}
+
+function os::test::extended::clusterup::verify_ruby_build () {
+    os::cmd::expect_success "oc new-app https://github.com/openshift/ruby-hello-world.git"
+    os::cmd::try_until_success "oc get builds -l app=ruby-hello-world | grep 'Complete'" $(( 10*minute )) 1
+}
+
+function os::test::extended::clusterup::verify_persistent_volumes () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get jobs -n default" "persistent-volume-setup"
+    os::cmd::try_until_success "oc get job persistent-volume-setup -n default -o jsonpath='{ .status.succeeded }' | grep 1" $(( 10*minute )) 1
+    os::cmd::expect_success "oc get pv/pv0100"
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_metrics () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get pods -n openshift-infra" "metrics-deployer"
+    os::cmd::try_until_success "oc get pods -n openshift-infra -l job-name=metrics-deployer-pod | grep Completed" $(( 10*minute )) 1
+    os::cmd::try_until_success "oc get endpoints -n openshift-infra"
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_logging () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get pods -n logging" "logging-deployer"
+    os::cmd::try_until_success "oc get pods -n logging -l logging-infra=deployer | grep Completed" $(( 10*minute )) 1
+    os::cmd::try_until_success "oc get endpoints logging-kibana -n logging | grep ':3000'" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::standard_test () {
+    local cmd="oc cluster up $@"
+    os::cmd::expect_success "${cmd}"
+    os::test::extended::clusterup::verify_router_and_registry
+    os::test::extended::clusterup::verify_image_streams
+    os::test::extended::clusterup::verify_ruby_build
+    os::test::extended::clusterup::verify_persistent_volumes
+}
+
+
+# Test with host dirs specified
+function os::test::extended::clusterup::internal::hostdirs () {
+    local data_dir="$(mktemp -d)"
```

Comment:
don't mix scoping and subshells
```sh
local data_dir; data_dir="$( mktemp -d )"
```

Also root these somewhere under `$BASETMPDIR`